hrm-service/migrations/20220415170355-create-Audits.js

```diff
@@ -0,0 +1,143 @@
+    console.log('Sequence "Audits_id_seq" ownership altered.');
+
+    await queryInterface.sequelize.query(
+      `CREATE TABLE IF NOT EXISTS public."Audits"
+      (
+        id integer NOT NULL DEFAULT nextval('"Audits_id_seq"'::regclass),
+        "tableName" text NOT NULL,
+        "targetPK" text NOT NULL,
+        operation text NOT NULL,
+        "updatedBy" text,
+        "previousRecord" jsonb,
+        timestamp_trx timestamp with time zone NOT NULL,
+        timestamp_stm timestamp with time zone NOT NULL,
+        timestamp_clock timestamp with time zone NOT NULL,
+        PRIMARY KEY (id)
+      );`,
+    );
+    console.log('Table "Audits" created');
+
+    await queryInterface.sequelize.query(`ALTER TABLE IF EXISTS public."Audits" OWNER TO hrm`);
+    console.log('Table "Audits" ownership altered.');
+
+    await queryInterface.sequelize.query(`
+      CREATE FUNCTION public.audit_trigger()
+        RETURNS trigger
+        LANGUAGE 'plpgsql'
+        NOT LEAKPROOF
+      AS $BODY$
+      DECLARE
+        audit_row public."Audits";
+        "targetPK" text;
+        "updatedBy" text;
+      BEGIN
+        IF TG_WHEN <> 'AFTER' THEN
+          RAISE EXCEPTION 'audit_trigger() may only run as an AFTER trigger';
+        END IF;
+
+        IF (TG_OP = 'UPDATE' AND TG_LEVEL = 'ROW') THEN
+          "updatedBy" = NEW."updatedBy";
+          IF (TG_TABLE_NAME = 'CaseSections') THEN
+            "targetPK" = concat(NEW."caseId", NEW."sectionType", NEW."sectionId");
+          ELSE
+            "targetPK" = NEW."id";
+          END IF;
+        ELSIF (TG_OP = 'INSERT' AND TG_LEVEL = 'ROW') THEN
+          "updatedBy" = NEW."createdBy";
+          IF (TG_TABLE_NAME = 'CaseSections') THEN
+            "targetPK" = concat(NEW."caseId", NEW."sectionType", NEW."sectionId");
+          ELSE
+            "targetPK" = NEW."id";
+          END IF;
+        ELSIF (TG_OP = 'DELETE' AND TG_LEVEL = 'ROW') THEN
+          "updatedBy" = NULL;
+          IF (TG_TABLE_NAME = 'CaseSections') THEN
+            "targetPK" = concat(OLD."caseId", OLD."sectionType", OLD."sectionId");
+          ELSE
+            "targetPK" = OLD."id";
+          END IF;
+        ELSE
+          RAISE EXCEPTION 'audit_trigger() added as trigger for unhandled case: %, %',TG_OP, TG_LEVEL;
+          RETURN NULL;
+        END IF;
+        
+        audit_row = ROW(
+          nextval('"Audits_id_seq"'::regclass), -- new audit id
+          TG_TABLE_NAME,                        -- target tabla name
+          "targetPK",                           -- target row primary key
+          TG_OP,                                -- operation performed on target row
+          "updatedBy",                          -- who's performing the above operation on the target row
+          to_jsonb(OLD),                        -- target record previous state
+          current_timestamp,                    -- transaction timestamp
```

Comment:
Like recording all these timestamps! This enhances the audit table as a troubleshooting resource 👍